### PR TITLE
Step7 Add New Controller and View Files for Task Management and Integrate RuboCop

### PR DESCRIPTION
**Pull request by fanfannb**
Description:
This pull request introduces a new controller and associated view files for managing tasks within the application.  Additionally, Rubocop has been integrated to enforce code quality, and some automatic corrections have been applied.

Changes:
Created a TasksController with actions for listing, showing, creating, editing, and deleting tasks.
Added corresponding view templates for each action, including index, show, new, and edit.
Integrated Rubocop for static code analysis and applied automatic corrections to the codebase.
These changes provide the foundational structure needed for task management within the application, enhancing functionality and improving code quality.

### ステップ7: タスクを登録・更新・削除できるようにしよう

- タスクの一覧画面、作成画面、詳細画面、編集画面を作成しましょう
  - `rails generate` コマンドでコントローラとビューを作成します
  - コントローラとビューに必要な実装を追加しましょう
  - 作成、更新、削除後はそれぞれflashメッセージを画面に表示させましょう
- `routes.rb` を編集して、 `http://localhost:3000/` でタスクの一覧画面が表示されるようにしましょう
- `Gemfile` で [RuboCop](https://github.com/rubocop/rubocop)（Rubyの静的コード解析ツール） か [fablicop](https://github.com/Fablic/fablicop)をインストールしましょう。設定やコマンド実行方法は各READMEから確認しましょう。
截屏2024-05-28 17 29 08
截屏2024-05-28 17 29 00
截屏2024-05-28 17 28 54

### CODE REVIEW

File: myapp/bin/webpack [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-ENV["NODE_ENV"]  ||= "development"
+ENV['RAILS_ENV'] ||= ENV['RACK_ENV'] || 'development'
+ENV['NODE_ENV']  ||= 'development'
 
-require "pathname"
-ENV["BUNDLE_GEMFILE"] ||= File.expand_path("../../Gemfile",
-  Pathname.new(__FILE__).realpath)
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile',
+                                           Pathname.new(__FILE__).realpath)
```
Comment (icer7) [inline myapp/bin/webpack:9]: nits
It doesn't affect the application itself, but I thought it would be a good idea to remove it because of the unnecessary whitespace!

File: myapp/bin/webpack-dev-server [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-ENV["NODE_ENV"]  ||= "development"
+ENV['RAILS_ENV'] ||= ENV['RACK_ENV'] || 'development'
+ENV['NODE_ENV']  ||= 'development'
 
-require "pathname"
-ENV["BUNDLE_GEMFILE"] ||= File.expand_path("../../Gemfile",
-  Pathname.new(__FILE__).realpath)
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile',
+                                           Pathname.new(__FILE__).realpath)
```
Comment (icer7) [inline myapp/bin/webpack-dev-server:9]: nits
It doesn't affect the application itself, but I thought it would be a good idea to remove it because of the unnecessary whitespace!

File: myapp/app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    if @task.save
+      flash[:notice] = 'Task was successfully created.'
+      redirect_to @task
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @task = Task.find(params[:id])
```
Comment (icer7) [inline myapp/app/controllers/tasks_controller.rb]:
```ruby
class TasksController < ApplicationController
  before_action :set_task, only: [:show, :edit, :update, :destroy]

  def show
    # @task is set by the before_action :set_task
  end

  def edit
    # @task is set by the before_action :set_task
  end

  def update
    if @task.update(task_params)
    ////
  end

  def destroy
    @task.destroy
     ////
  end

  private

  def set_task
    @task = Task.find(params[:id])
  end
```

Perhaps it is a matter of taste, but I thought it would be a good idea to put them together since there is some duplication of processing!

**Review by icer7 — APPROVED** [review]
FYI なので修正必須ではないです！

File: myapp/app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -7,9 +7,18 @@
 
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <!-- Bootstrap CSS -->
+    <link href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" rel="stylesheet">
```
Comment (nagamine-r) [inline myapp/app/views/layouts/application.html.erb]: Why 4.5.2? If you want to use a latest version of 4.x.x then you should use v4.6.2.
Or, you want to use a latest version of bootstrap then you should use v5.3.2.
https://www.bootstrapcdn.com/

File: myapp/app/views/tasks/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  <%= link_to 'New Task', new_task_path, class: 'btn btn-primary mb-3' %>
+
+  <div class="row">
+    <% @tasks.each do |task| %>
+      <div class="col-md-3"> <!-- 调整列的宽度 -->
+        <div class="card mb-4" style="height: 250px;"> <!-- 调整卡片的高度 -->
+          <div class="card-body d-flex flex-column">
+            <h5 class="card-title">Title: <%= link_to task.title, task %></h5> <!-- 添加链接到 show 页面 -->
+            <p class="card-text" style="flex-grow: 1; max-height: 100px; overflow: hidden;">Description: <%= task.description %></p>
```
Comment (nagamine-r) [inline myapp/app/views/tasks/index.html.erb]: I think you can use it.
https://getbootstrap.jp/docs/5.3/utilities/flex/#%e6%8b%a1%e5%a4%a7%e3%81%a8%e7%b8%ae%e5%b0%8f
```suggestion
            <p class="card-text flex-grow-1" style="max-height: 100px; overflow: hidden;">Description: <%= task.description %></p>
```

File: myapp/app/views/tasks/index.html.erb [hunk 6]
```diff
@@ -3,23 +3,20 @@
...
-          <td><%= task.description %></td>
-          <td><%= link_to 'Edit', edit_task_path(task), class: 'btn btn-warning btn-sm' %></td>
-          <td><%= button_to 'Delete', task, method: :delete, class: 'btn btn-danger btn-sm', data: { confirm: 'Are you sure?' } %></td>
-        </tr>
-      <% end %>
-    </tbody>
-  </table>
+  <div class="row">
+    <% @tasks.each do |task| %>
+      <div class="col-md-3"> <!-- 调整列的宽度 -->
```
Comment (nagamine-r) [inline myapp/app/views/tasks/index.html.erb]: FYI: I'd prefer you to comment in English, normally. But that's OK, because this is an training project.

**Review by nagamine-r — APPROVED** [review]
I commented, but LGTM.

File: myapp/bin/webpack (hunk 1 above)
Comment (fanfannb) [reply myapp/bin/webpack:9]: If I delete the white space, it will change back when I use rubocop to automatically correct it. So I didn't change here

File: myapp/app/controllers/tasks_controller.rb (hunk 3 above)
Comment (fanfannb) [reply myapp/app/controllers/tasks_controller.rb]: Thank you！ I changed it

File: myapp/app/views/layouts/application.html.erb (hunk 4 above)
Comment (fanfannb) [reply myapp/app/views/layouts/application.html.erb]: Thank you. I've made it the latest version！

File: myapp/app/views/tasks/index.html.erb (hunk 5 above)
Comment (fanfannb) [reply myapp/app/views/tasks/index.html.erb]: Thank you！ I changed it
、

File: myapp/bin/webpack (hunk 1 above)
Comment (icer7) [reply myapp/bin/webpack:9]: Oh, I see.
There was a word limit within a line.
My apologies!

File: myapp/app/views/tasks/index.html.erb (hunk 6 above)
Comment (fanfannb) [reply myapp/app/views/tasks/index.html.erb]: Thank you！ I changed it

File: myapp/app/controllers/tasks_controller.rb [hunk 7]
```diff
@@ -1,12 +1,14 @@
 # frozen_string_literal: true
 
 class TasksController < ApplicationController
+  before_action :set_task, only: [:show, :edit, :update, :destroy]
```
Comment (icer7) [inline myapp/app/controllers/tasks_controller.rb:4]: 👍

**Review by icer7 — APPROVED** [review]
LGTM!
